File: code-coaches/medium/text-decompressor/text_decompressor.py
def decompress_text(compressed_text: str) -> str:
    decompressed_text = ""
    for idx in range(1, len(compressed_text), 2):
        decompressed_text += compressed_text[idx - 1] * int(compressed_text[idx])

    return decompressed_text


def main():
    compressed_text = input()
    print(decompress_text(compressed_text))


if __name__ == "__main__":
    main()

File: code-coaches/medium/text-decompressor/TextDecompressor.swift
func decompressText(compressedText: String) -> String {
    var decompressedText = ""
    var idx = 0
    while idx < compressedText.count {
        let character = compressedText[compressedText.index(compressedText.startIndex, offsetBy: idx)]
        let repeatCount = Int(String(compressedText[compressedText.index(compressedText.startIndex, offsetBy: idx + 1)])) ?? 0
        decompressedText += String(repeating: String(character), count: repeatCount)
        idx += 2
    }
    return decompressedText
}

func main() {
	print(decompressText(compressedText: readLine()!))
}

main()
